File: learning_management_app/lib/models/student/teacher_chat_response.dart
class TeacherChatResponse {
  final int teacherId;
  final String teacherName;
  final String departmentName;
  final String email;

  TeacherChatResponse({
    required this.teacherId,
    required this.teacherName,
    required this.departmentName,
    required this.email,
  });

  factory TeacherChatResponse.fromJson(Map<String, dynamic> json) {
    return TeacherChatResponse(
      teacherId: json['teacherId'] ?? 0,
      teacherName: json['teacherName'] ?? '',
      departmentName: json['departmentName'] ?? '',
      email: json['email'] ?? '',
    );
  }
}

File: LearningManagementBackend/src/main/java/org/learn/learningmanagementbackend/dto/response/TeacherChatResponse.java
package org.learn.learningmanagementbackend.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class TeacherChatResponse {
    private Integer teacherId;
    private String teacherName;
    private String departmentName;
    private String email;
}
